tensorflow/tools/xla_extract/tf_graph_to_xla_lib.cc

```diff
@@ -111,47 +122,47 @@ xla::HloModuleProto ExtractHloFromGraphDef(const GraphDef& in_graph,
   }
   std::unique_ptr<ClientGraph> client_graph;
   s = execution_state->BuildGraph(bg_options, &client_graph);
-  if (!s.ok()) LOG(FATAL) << "build graph failed " << s.error_message();
+  if (!s.ok()) LOG(ERROR) << "build graph failed " << s.error_message();
 
   // Usually there is only one cluster, but for some graphs (e.g. LSTM) there
   // may be more.  Return the *last* cluster whose name starts with "cluster_"
   FunctionDefLibrary fdef_lib = client_graph->flib_def->ToProto();
 
   auto fdef_iter =
-      std::find_if(fdef_lib.function().rbegin(), fdef_lib.function().rend(),
+      std::find_if(fdef_lib.mutable_function()->rbegin(), fdef_lib.mutable_function()->rend(),
                    [](const FunctionDef& f_) -> bool {
                      return (f_.signature().name().find("cluster_") == 0 &&
                              f_.signature().name().substr(
                                  f_.signature().name().length() - 2) == "_0");
                    });
 
-  FunctionDef fdef;
-
-  if (fdef_iter == fdef_lib.function().rend()) {
+  if (fdef_iter == fdef_lib.mutable_function()->rend()) {
     fdef_iter =
-        std::find_if(fdef_lib.function().rbegin(), fdef_lib.function().rend(),
+        std::find_if(fdef_lib.mutable_function()->rbegin(), fdef_lib.mutable_function()->rend(),
                      [](const FunctionDef& f_) -> bool {
                        return (f_.signature().name().find("cluster_") == 0);
                      });
   }
 
-  if (fdef_iter != fdef_lib.function().rend()) {
-    fdef = *fdef_iter;
-  } else {
-    fdef = *fdef_lib.function().begin();
-    LOG(INFO) << "cluster not found, using " << fdef.signature().name()
-              << " instead\n";
+  if (fdef_iter == fdef_lib.mutable_function()->rend()) {
+    fdef_iter = fdef_lib.mutable_function()->rend()-1;
```

Comment:
why not --fdef_iter?
tbh, I never saw anyone adjust against an end iter.  I guess it's legal?  What is the container is empty? Is that handled before here?

Reply:
the container cant be empty.   yeah its legal :P